presto-base-arrow-flight/src/main/java/com/facebook/plugin/arrow/ArrowFlightRequest.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.plugin.arrow;
+
+public interface ArrowFlightRequest
```

Comment:
Can you explain why this is need instead of just using a `FlightDescriptor`?

Reply:
Since flight server implementations can accept command in their own format, we created this interface that can be implemented by connector implementations to build the command in the format needed. https://github.com/sabbasani/presto/blob/arrow-connector/presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestingArrowFlightRequest.java This implementation returns the command in a JSON format.

Reply:
Right, but normally you would define a model for the command, like JSON or protobuf etc, which is serialized to bytes. Then those bytes are just added to the FlightDescriptor, there isn't a need for an additional interface.

Reply:
+1